classes/NewsFetcher.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+use Doctrine\Common\Cache\FilesystemCache;
+use GuzzleHttp\Message\Request;
+use GuzzleHttp\Subscriber\Cache\CacheStorage;
+use GuzzleHttp\Subscriber\Cache\CacheSubscriber;
+use PrestaShop\CircuitBreaker\AdvancedCircuitBreakerFactory;
+use PrestaShop\CircuitBreaker\Contract\FactoryInterface;
+use PrestaShop\CircuitBreaker\FactorySettings;
+use PrestaShop\CircuitBreaker\Storage\DoctrineCache;
+
+class NewsFetcherCore
```

Comment:
We want to get rid of the object model classes (all classes in the `classes` directory) It is some legacy code that we are stuck with, so we don't have a choice for now
But let's avoid adding some in it ^^ Besides object model classes are usually linked to persisted data in the database which is not the case here
And the `ClassnameCore` convention of these objects is used to allow the infamous *override* feature in PrestaShop which we also want to get rid of (or at least not encourage for now ^^)

So in this case you'd better create a service, and you can divide it into multiple sub services:
```yaml
# circuit_breaker.yml
services:
    prestashop.core.circuit_breaker.doctrine_cache:
        class: Doctrine\Common\Cache\FilesystemCache
        arguments:
            - "@=service('prestashop.adapter.environment').getCacheDir()/circuit_breaker"

    prestashop.core.circuit_breaker.guzzle.cache_storage:
        class: GuzzleHttp\Subscriber\Cache\CacheStorage
        arguments:
             - "@prestashop.core.circuit_breaker.doctrine_cache"
             - 86400
    
    prestashop.core.circuit_breaker.guzzle.cache_subscriber:
        class: GuzzleHttp\Subscriber\Cache\CacheSubscriber

    prestashop.core.circuit_breaker.storage:
        class: PrestaShop\CircuitBreaker\Storage\DoctrineCache
        arguments:
            - "@prestashop.core.circuit_breaker.doctrine_cache"

    prestashop.core.circuit_breaker.advanced_factory:
        class: PrestaShop\CircuitBreaker\AdvancedCircuitBreakerFactory
```

These are the basic services you will need to create different instances of circuit breakers. Maybe they will we needed in legacy environment, not just Symfony so be careful where you define them (more details about service scopes here: https://devdocs.prestashop.com/1.7/modules/concepts/services/#services-in-legacy-environment

Then once you have those basics you can manage your circuit breaker settings via services as well:

```yaml
# news_services.yml
services:
    prestashop.core.news.circuit_breaker.settings:
        class: PrestaShop\CircuitBreaker\FactorySettings
        # Here I use hard coded values but they should be in a class constants
        arguments:
            - 3
            - 3
            - 0
        calls:
            - ['setThreshold', [86400]]
            - ['setStrippedFailures', [3]]
            - ['setStrippedTimeout', [3]]
            - ['setStorage', ['@prestashop.core.circuit_breaker.storage']]
            - ['setClientOptions', ['method': 'GET', 'subscribers': ['@prestashop.core.circuit_breaker.guzzle.cache_subscriber']]]

    prestashop.core.news.circuit_breaker:
        factory: ["@prestashop.core.circuit_breaker.advanced_factory", create]
        arguments:
            - "@prestashop.core.news.circuit_breaker.settings"

    prestashop.core.news.fetcher:
        class: "RightNamespace\NewsFetcher"
        arguments:
            - "@prestashop.core.news.circuit_breaker"
```

Y hasta!! Now all you need to do is create your `NewsFetcher` service class (which you already have) without caring about the circuit breaker instanciation and focusing on parsing your datas. And this will be easier to unit test as well

Reply:
Done